smolyay/samples.py

```diff
@@ -0,0 +1,409 @@
+import abc
+import numpy
+
+
+class UnidimensionalPointSet(abc.ABC):
+    """Set of unidimensional points
+
+    A set of unique unidimensional points within a domain.
+    """
+
+    def __init__(self, domain):
+        self._points = None
+        self._domain = numpy.array(domain)
```

Comment:
You should make sure to cast the type, in case they are given as ints or something.
```suggestion
        self._domain = numpy.array(domain, dtype=float)
```